ERCS/erc-7715.md

```diff
@@ -216,21 +198,23 @@ type WalletSigner = {
 };
 
 // A signer representing a single key.
-// `id` is a did:key identifier and can therefore represent both Secp256k1 or Secp256r1 keys, among other key types.
+// "Key" types are explicitly secp256r1 (p256) or secp256k1, and the public keys are hex-encoded.
 type KeySigner = {
   type: "key";
   data: {
-    id: string;
+    type: "secp256r1" | "secp256k1";
+    publicKey: `0x${string}`;
   };
 };
 
 // A signer representing a multisig signer.
-// Each element of `ids` is a did:key identifier just like the `key` signer.
+// Each element of `data` is explicitly an secp256r1 (p256) or secp256k1 key, and the public keys are hex-encoded
 type MultiKeySigner = {
   type: "keys";
   data: {
-    ids: string[];
-  };
+    type: "secp256r1" | "secp256k1";
+    publicKey: `0x${string}`;
+  }[];
```

Comment:
forgot exactly where we landed on this, does this seem right? @pedrouid @lukaisailovic

Reply:
@lukasrosario Each signer has `type` and `data` fields. We can't make `data` into array.
Proposing that we do this:
```
data: {
keys: {
type: ...
publicKey: ...
}[]
}
```

wdyt?

Reply:
this assumes all the keys would be the same type. are there use cases for mixing the types? if not this is fine with me

Reply:
@lukasrosario Our use case involves multiple keys with different types. Also all other signers have `data`, not `data[]`

Reply:
how does what you wrote cover that use case then? there is only one `type`

Reply:
i updated it, lmk if i misunderstood

Reply:
Can't add suggestion, pasting it here. Lets do this:
```
data: {
  keys: {
    type: KeyType;
    publicKey: `0x${string}`
  }[];
};
```

Reply:
ah i see. misread your initial comment my bad

Reply:
updated!